From f9bc85775b2f3f3c1ff37b48072c8cd771eec1c8 Mon Sep 17 00:00:00 2001
From: Brett K <brettk@groupon.com>
Date: Thu, 20 Jun 2013 14:18:42 -0500
Subject: [PATCH 1/1] Skip package prune if dry run

---
 src/roll.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/roll.c b/src/roll.c
index 17bb63b..16435e0 100644
--- a/src/roll.c
+++ b/src/roll.c
@@ -726,7 +726,10 @@ int main(int argc, char *argv[]) {
         clean_previous_package_trees(package_target_dir,
                                      options.dryrun ? temp_package_link_dir : package_link_dir,
                                      previous_package_link_dir);
-        if(prune_packages) {
+        if(options.dryrun) {
+          log_info("Skipping package prune in dry run mode");
+        }
+        else if (prune_packages) {
           log_info("Removing unused packages from prior installations.");
           clean_previous_packages(hostclass_config.package_list,
                                   package_stow_dir);
-- 
1.8.0

